test/extensions/http/header_formatters/preserve_case/preserve_case_formatter_reason_phrase_integration_test.cc

```diff
@@ -0,0 +1,70 @@
+#include "envoy/extensions/http/header_formatters/preserve_case/v3/preserve_case.pb.h"
+
+#include "test/integration/filters/common.h"
+#include "test/integration/http_integration.h"
+#include "test/test_common/registry.h"
+
+namespace Envoy {
+namespace {
+
+class PreserveCaseFormatterReasonPhraseIntegrationTest
+    : public testing::TestWithParam<Network::Address::IpVersion>,
+      public HttpIntegrationTest {
+public:
+  PreserveCaseFormatterReasonPhraseIntegrationTest()
+      : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {}
+
+  void SetUp() override {
+    setDownstreamProtocol(Http::CodecType::HTTP1);
+    setUpstreamProtocol(Http::CodecType::HTTP1);
+  }
+
+  void initialize() override {
+    if (upstreamProtocol() == Http::CodecType::HTTP1) {
+      config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+        ConfigHelper::HttpProtocolOptions protocol_options;
+        auto typed_extension_config = protocol_options.mutable_explicit_http_config()
+                                          ->mutable_http_protocol_options()
+                                          ->mutable_header_key_format()
+                                          ->mutable_stateful_formatter();
+        typed_extension_config->set_name("preserve_case");
+
+        auto config =
+            TestUtility::parseYaml<envoy::extensions::http::header_formatters::preserve_case::v3::
+                                       PreserveCaseFormatterConfig>("forward_reason_phrase: true");
+        typed_extension_config->mutable_typed_config()->PackFrom(config);
+
+        ConfigHelper::setProtocolOptions(*bootstrap.mutable_static_resources()->mutable_clusters(0),
+                                         protocol_options);
+      });
+    }
+
+    HttpIntegrationTest::initialize();
+  }
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, PreserveCaseFormatterReasonPhraseIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+TEST_P(PreserveCaseFormatterReasonPhraseIntegrationTest, VerifyReasonPhrase) {
```

Comment:
Can you add an off test also where when explicitly disabled you get back the stock phrase?

Reply:
Done!